Gems/Terrain/Code/Source/Components/TerrainPhysicsColliderComponent.cpp

```diff
@@ -456,22 +534,27 @@ namespace Terrain
 
     void TerrainPhysicsColliderComponent::GetHeightfieldGridSize(int32_t& numColumns, int32_t& numRows) const
     {
+        AZStd::shared_lock lock(m_stateMutex);
         numColumns = aznumeric_cast<int32_t>(m_heightfieldRegion.m_numPointsX);
         numRows = aznumeric_cast<int32_t>(m_heightfieldRegion.m_numPointsY);
     }
 
     int32_t TerrainPhysicsColliderComponent::GetHeightfieldGridColumns() const
     {
+        AZStd::shared_lock lock(m_stateMutex);
         return aznumeric_cast<int32_t>(m_heightfieldRegion.m_numPointsX);
     }
 
     int32_t TerrainPhysicsColliderComponent::GetHeightfieldGridRows() const
     {
+        AZStd::shared_lock lock(m_stateMutex);
         return aznumeric_cast<int32_t>(m_heightfieldRegion.m_numPointsY);
     }
```

Comment:
I'm having a hard time understanding how the locks help here. Say something was calling
```
GetHeightfieldGridSize()
GetHeightfieldGridColumns()
GetHeightfieldGridRows()
```
and something happened to call `CalculateHeightfieldRegion()` right in the middle of it. Well, you'll still get some values that are returned from the old version of m_heightFieldRegion, and some from the new version.

I can maybe see a case for the lock on `GetHeightfieldGridSize()` because then the rows / columns will be guaranteed to be in sync if an update happens in the middle, but not on the single property returns.

Reply:
Looks like formally this might be UB but practically on a 64 bit machine the size_t would probably be safe? I'm fine with keeping it.

Reply:
Yeah, I'm a fan of ensuring atomic operation correctness instead of relying on architecture and implementation assumptions. :) If these calls become more performance-critical, one of the first things I'd do is combine them together anyways, since the most common use case is to need multiple configuration values.